### DISCUSSION TITLE
problems with running toga-demo

### DISCUSSION THREAD

**Discussion by icrabbe** [beeware/toga#1625] [General]
I am just startng out with toga. I want to use it to build a cross-platform app. while going through the quick start I type 
(venv) (base) ira@LAPTOP-TU41MCKE:/mnt/c/Users/iacra/helloworld$ cd toga
(venv) (base) ira@LAPTOP-TU41MCKE:/mnt/c/Users/iacra/helloworld/toga$ pwd
/mnt/c/Users/iacra/helloworld/toga
(venv) (base) ira@LAPTOP-TU41MCKE:/mnt/c/Users/iacra/helloworld/toga$ pip install -e src/core
ERROR: src/core is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file).
(venv) (base) ira@LAPTOP-TU41MCKE:/mnt/c/Users/iacra/helloworld/toga$ pip install -e src/dummy
ERROR: src/dummy is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file).
(venv) (base) ira@LAPTOP-TU41MCKE:/mnt/c/Users/iacra/helloworld/toga$ ls
README.md  __pycache__  browser.py  canvas.py  helloworld.py  multiplebox.py  temperatureconverter.py
What is causing this error?
Ira C

**Comment by freakboy3742** [comment]
The error is being caused because you're trying to install something that doesn't exist. You've `cd`'ed into *a* `toga` directory, but that directory clearly isn't a checkout of the Toga source tree, because there isn't a `src` folder under that directory, but there *is* a range of other code... none of which I can work out where it came from. As a result, `pip install` fails... because the thing you're asking to install (`src/core`) doesn't exist.

It's not clear which instructions you're following, because neither the [BeeWare tutorial](https://docs.beeware.org), nor the [Toga tutorial](https://toga.readthedocs.io/en/latest/tutorial/index.html#tutorial) would instruct you to run the commands you've described. The instructions you've described are *roughly* consistent with the *developers* guide - instructions for someone who wants to set up Toga so they can make a contribution to the project - but you've missed a bunch of steps, and obtained some source code from *somewhere* that doesn't look at all familiar.
